### PR TITLE
ZCS-1182: Universal Ui - Calendar main views left nav and action bar.

### PR DESCRIPTION
**Pull request by tarangkhandelwal**
ZCS-1182: Universal Ui - Calendar main views left nav and action bar.

Changeset:
 * ZmCalViewController.js:
     * Updating navigation title format for month view to full month name.
     * Adding tag menu & print button to actions menu using _getSecondaryToolBarOps method of base controller.
     * Overriding ZmBaseController._getSecondaryToolBarOpsPosition() to return index of actions menu as 2.
     * Overriding ZmBaseController._getButtonOverrides() to override calendar view operations & adding a specific class to them to style them look as tabs.
* ZmCalViewController.js: Using AjxMsg.formatDateMediumNoYear to update the navbar title for week views.
* ZmCalMonthView.js: Adding method getCalTitle to return full month name using DwtCalendar.getMonthFormatter.
* ZmBaseController.js: Adding method _getSecondaryToolBarOpsPosition that can be used to return the index where actions bar should be placed in the toolbar.
* ZmButtonToolbar.js: Changes to respect the actions menu position and arrange the operations based on the positions.
* ZmToolBar.js: Correcting the way the className is passed. The function is called from the same file where className is sent inside params.

### CODE REVIEW

File: WebRoot/css/dwt.css [hunk 1]
```diff
@@ -2196,6 +2196,9 @@ INPUT.xform_field {
 
 .ZTab									{	@Tab@					}
 .ZTabPage								{	@TabPage@ 				}
+.ZTabButton								{	@TabButton@				}
+.ZSelected.ZTabButton					{	@TabButton-selected@	}
```
Comment (silentsakky) [inline WebRoot/css/dwt.css]: minor: indentation

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js [hunk 2]
```diff
@@ -1032,24 +1030,59 @@ function(items) {
...
+		ZmOperation.WEEK_VIEW,
 		ZmOperation.MONTH_VIEW,
-        ZmOperation.CAL_LIST_VIEW
+		ZmOperation.CAL_LIST_VIEW
 	];
-    if( appCtxt.get(ZmSetting.FREE_BUSY_VIEW_ENABLED) ){
-        toolbarOptions.push(ZmOperation.FB_VIEW);
-    }
-    return toolbarOptions;
+		if( appCtxt.get(ZmSetting.FREE_BUSY_VIEW_ENABLED) ){
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js]: minor: indentation

File: WebRoot/js/zimbraMail/share/controller/ZmBaseController.js [hunk 3]
```diff
@@ -256,6 +256,9 @@ ZmBaseController.prototype._getToolBarOps 			= function() {};
 // Returns a list of secondary (non primary) toolbar operations
 ZmBaseController.prototype._getSecondaryToolBarOps 	= function() {};
 
+//Returns the index where secondary buttons action menu should be places
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/share/controller/ZmBaseController.js]: minor: should be placed

File: WebRoot/skins/_base/base4/skin.css [hunk 4]
```diff
@@ -758,6 +758,14 @@ we are hiding element with that class
     fill: @IconCheckboxUnchecked@;
 }
 
+.ZTreeItemCheckbox .ImgCheckboxChecked-svg .svg-icon {
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.css]: can we use variables like
CheckElmCheckBgColor
CheckElmUncheckBgColor

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js [hunk 5]
```diff
@@ -1548,7 +1581,7 @@ function(date, duration, roll) {
 			this.show(ZmId.VIEW_CAL_WEEK);
 		}
 		if (ZmId.VIEW_CAL_MONTH == this._currentViewType) {
-			title = this._viewMgr.getCurrentView().getShortCalTitle();
+			title = this._viewMgr.getCurrentView().getCalTitle();
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js]: i think we can get rid of this if condition
as we are already getting title from current view on line 1541 (currentView.getCalTitle())

File: WebRoot/css/dwt.css (hunk 1 above)
Comment (tarangkhandelwal) [inline WebRoot/css/dwt.css]: its tab based indentation. We can ignore this as discussed.

File: WebRoot/skins/_base/base4/skin.css (hunk 4 above)
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.css]: Was thinking not to have variables for them as we didn't had them for any of the svg icons (above & below this line). Tried to just refer to the color we have in a variable for the checkboxes. 

Let me know your thoughts.

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js (hunk 5 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js]: Removed here : 0e0e88c9e3034ae1b51f6657d6aaf2d93a0c4a03
Thanks

File: WebRoot/skins/_base/base4/skin.css (hunk 4 above)
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.css]: Updated the rules as per offline discussion:
0e0e88c9e3034ae1b51f6657d6aaf2d93a0c4a03

File: WebRoot/js/zimbraMail/share/controller/ZmBaseController.js (hunk 3 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/share/controller/ZmBaseController.js]: fixed: 0e0e88c9e3034ae1b51f6657d6aaf2d93a0c4a03

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js (hunk 2 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js]: Fixed:
0e0e88c9e3034ae1b51f6657d6aaf2d93a0c4a03

File: WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js [hunk 6]
```diff
@@ -437,7 +437,7 @@ function(hour) {
 
 ZmCalColView.prototype._updateTitle =
 function() {
-	var dayFormatter = DwtCalendar.getDayFormatter();
+	var dayFormatter = new AjxDateFormat(AjxMsg.formatDateMediumNoYear);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js]: Can we create a method in DwtCalendar to cache formatter object as was done previously

File: WebRoot/skins/_base/base4/skin.properties [hunk 7]
```diff
@@ -1682,7 +1684,7 @@ ChooserDialogBody                    = width:320px; @Text@
 ChooserDialogInputContainer          = padding-bottom:3px; padding-top:3px;
 ChooserDialogInputField              = @borderRadius@:2px; @PageBg@; @SeparatorBorder@; width:320px; height:32px; margin-bottom:16px;
 ChooserDialogItemContainer           = background-color:@AppC@; overflow:auto;
-ChooserDialogItemInnerContainer      = @SeparatorBorder@; padding:18px 0px 0px 16px; border-color:@SeparatorColor@; @borderRadius@:2px;
+ChooserDialogItemInnerContainer      = @SeparatorBorder@; border-color:@SeparatorColor@; @borderRadius@:2px;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: do we need border-color here? @SeparatorBorder@ internally uses @SeparatorColor@

Comment (tarangkhandelwal) [reply WebRoot/skins/_base/base4/skin.properties]: Removed

File: WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js (hunk 6 above)
Comment (tarangkhandelwal) [reply WebRoot/js/zimbraMail/calendar/view/ZmCalColView.js]: Using AjxDateUtil._dateFormatNoYear; now.

**Pull request merged**